editor/store/reducer.js

```diff
@@ -114,7 +117,27 @@ export const editor = flow( [
 	combineReducers,
 
 	// Track undo history, starting at editor initialization.
-	partialRight( withHistory, { resetTypes: [ 'SETUP_NEW_POST', 'SETUP_EDITOR' ] } ),
+	partialRight( withHistory, {
+		resetTypes: [ 'SETUP_NEW_POST', 'SETUP_EDITOR' ],
+		shouldOverwriteState( action, previousAction ) {
+			if ( ! includes( [ 'UPDATE_BLOCK_ATTRIBUTES', 'EDIT_POST', 'RESET_POST' ], action.type ) ) {
+				return false;
+			}
+
+			if (
+				previousAction &&
+				action.type === 'UPDATE_BLOCK_ATTRIBUTES' &&
+				action.type === previousAction.type
+			) {
+				const attributes = keys( action.attributes );
+				const previousAttributes = keys( previousAction.attributes );
+
+				return action.uid === previousAction.uid && isEqual( attributes, previousAttributes );
+			}
+
+			return true;
```

Comment:
Don't you think it should be false by default? and we'd only overwrite the state if the action is `UPDATE_BLOCK_ATTRIBUTES` with the same keys?

Reply:
This would allow us to remove the check you're doing line 123

Reply:
Hm, then I just have to invert that check?

Reply:
yes, why can't this function be just:

```js
shouldOverwriteState( action, previousAction ) {
			if (
				previousAction &&
				action.type === 'UPDATE_BLOCK_ATTRIBUTES' &&
				action.type === previousAction.type
			) {
				const attributes = keys( action.attributes );
				const previousAttributes = keys( previousAction.attributes );

				return action.uid === previousAction.uid && isEqual( attributes, previousAttributes );
			}

			return false;
},
```

```

Reply:
It still has checks for other actions too?

Reply:
no, why do we need to check other actions?

Reply:
That's why I commented here https://github.com/WordPress/gutenberg/pull/4956#discussion_r167826616. I'm having problems with loading the post and autosave interfering with the undo levels.

Reply:
Oh! I missed that one sorry.

Reply:
THinking about these actions 'EDIT_POST', 'RESET_POST'

I believe these two actions shouldn't overwrite previous state. If you edit a category, a publish date or anything in the post, it should create an undo level right? 

same for `reset_post` which means the post is updated globally somewhere.

Maybe we could add a flag to this action when we want to avoid creating undo levels (like post success actions)?

Reply:
Yep, it definitely a to do.